pymongo/collection.py

```diff
@@ -116,7 +117,7 @@ def __init__(
         database: "Database[_DocumentType]",
         name: str,
         create: Optional[bool] = False,
-        codec_options: Optional[CodecOptions] = None,
+        codec_options: Optional["CodecOptions[Mapping[str, Any]]"] = None,
```

Comment:
Do we need to use a generic TypeVar for CodecOptions instead of hardcoding `Mapping[str, Any]` to avoid this failure?:
```
Found 1 error in 1 file (checked 83 source files)
pymongo/mongo_client.py:2054: error: Argument "codec_options" to "get_database"
of "MongoClient" has incompatible type "CodecOptions[MutableMapping[str, Any]]";
expected "Optional[CodecOptions[Mapping[str, Any]]]"  [arg-type]
                codec_options=DEFAULT_CODEC_OPTIONS,
                              ^~~~~~~~~~~~~~~~~~~~~
```

Reply:
Actually we can use `[_DocumentType]` everywhere and satisfy `pyright` in strict mode.